client/components/app.jsx

```diff
@@ -22,18 +25,45 @@ class App extends React.Component {
         this.setState({
           grades: grade
         });
+        this.calculateAverageGrade();
       });
   }
 
+  // componentDidUpdate() {
+  //   this.calculateAverageGrade()
+  // }
+
+  calculateAverageGrade() {
+    let totalSumOfGrades = 0;
+    let averageGrade = 0;
+    const gradesLength = this.state.grades.length;
+    if (gradesLength) {
+      for (let i = 0; i < gradesLength; i++) {
+        totalSumOfGrades += this.state.grades[i].grade;
+      }
+      averageGrade = Math.floor(totalSumOfGrades / gradesLength);
+      this.setState(state => ({
+        average: averageGrade
+      }));
```

Comment:
The average is computed from something already in state, so there's not reason to store it in state. Only `return` the average.